libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationOwner.kt

```diff
@@ -60,6 +60,8 @@ internal class MapboxNavigationOwner {
             attached = false
             services.forEach { it.onDetached(mapboxNavigation!!) }
             MapboxNavigationProvider.destroy()
+            mapboxNavigation = null
```

Comment:
This is repeated in `carAppLifecycleObserver.onStop`, looks like it should be moved to a separate function.

Reply:
there are still some slight differences like logs and the condition for if (attached)

think i prefer to let them continue to be separate, cover the issues with tests